x/cheqd/keeper/msg_server_deactivate_did.go

```diff
@@ -0,0 +1,99 @@
+package keeper
+
+import (
+	"context"
+
+	"github.com/cheqd/cheqd-node/x/cheqd/types"
+	"github.com/cheqd/cheqd-node/x/cheqd/utils"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+const DeactivatedPostfix string = "-deactivated"
+
+func (k msgServer) DeactivateDid(goCtx context.Context, msg *types.MsgDeactivateDid) (*types.MsgDeactivateDidResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	// Validate DID does exist
+	if !k.HasDid(&ctx, msg.Payload.Id) {
+		return nil, types.ErrDidDocNotFound.Wrap(msg.Payload.Id)
+	}
+
+	// Validate namespaces
+	namespace := k.GetDidNamespace(ctx)
+	err := msg.Validate([]string{namespace})
+	if err != nil {
+		return nil, types.ErrNamespaceValidation.Wrap(err.Error())
+	}
+
+	// Retrieve existing state value and did
+	existingStateValue, err := k.GetDid(&ctx, msg.Payload.Id)
+	if err != nil {
+		return nil, err
+	}
+
+	existingDid, err := existingStateValue.UnpackDataAsDid()
+	if err != nil {
+		return nil, err
+	}
+
+	updatedMetadata := *existingStateValue.Metadata
+	updatedMetadata.Update(ctx)
+	updatedMetadata.Deactivated = true
+
+	updatedStateValue, err := types.NewStateValue(existingDid, &updatedMetadata)
+	if err != nil {
+		return nil, err
+	}
+
+	// Consider did that we are going to create during did resolutions
+	inMemoryDids := map[string]types.StateValue{existingDid.Id: updatedStateValue}
+
+	// Check controllers' existence
+	controllers := existingDid.AllControllerDids()
+	for _, controller := range controllers {
+		_, err := MustFindDid(&k.Keeper, &ctx, inMemoryDids, controller)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	// Verify signatures
+	updatedDid, err := updatedStateValue.UnpackDataAsDid()
+	if err != nil {
+		return nil, err
+	}
+	signers := GetSignerDIDsForDIDDeactivation(*existingDid, *updatedDid)
```

Comment:
Why does this method use 2 params. Only existingDid is needed. `GetSignerDIDsForDIDDeactivation` should look like  `GetSignerDIDsForDIDCreation`